iam-login-service/src/main/java/it/infn/mw/iam/core/oauth/profile/common/BaseAccessTokenBuilder.java

```diff
@@ -170,4 +182,20 @@ protected JWTClaimsSet.Builder baseJWTSetup(OAuth2AccessTokenEntity token,
     return builder;
   }
 
+  protected void addAcrClaimIfNeeded(Builder builder, OAuth2Authentication authentication) {
+
+    Optional<IamAccount> account = accountUtils.getAuthenticatedUserAccount(authentication);
+    if (authentication.getUserAuthentication() != null
+        && authentication.getUserAuthentication().getDetails() instanceof Map<?, ?> details
+        && details.get("acr") != null) {
+      builder.claim("acr", details.get("acr"));
+    } else {
+      if (account.isPresent()) {
+        Optional<IamTotpMfa> totpMfaOptional = totpMfaRepository.findByAccount(account.get());
+        if (totpMfaOptional.isPresent() && totpMfaOptional.get().isActive()) {
+          builder.claim("acr", "https://referds.org/profile/MFA");
```

Comment:
Could we have a case when we don't want to stress users to do the 2FA but we require this from the remote SSO? And we want to fail in case is not applied? Probably this will be required by someone. What about having the "default acr value to be required" as a configuration parameter for the remote OIDC provider?

Reply:
We have already discussed this verbally, here we are just building the access token by inserting the `acr` claim